### PR TITLE
Complete Masterclass Refresh

### LINKED ISSUE #34: Issues with step 5.2
I'm seeing two issues in step 5.2 - One appears to be a syntax issue that just requires the removal of the `--app` option in:

```
balena config generate --app cliApp --version 2.38.0+rev1 --device 6053dab8dc4721ed288c8dfc79e52967 --network ethernet --appUpdatePollInterval 10 --output config.json
--device= cannot also be provided when using --app=
```

It executes fine without `--app cliApp`

The next command however gives an error after prompting for the network connection - Here's the output with debug on:

```
balena os configure balena-rpi4-image.img --config config.json --device 6053dab8dc4721ed288c8dfc79e52967 --deb
ug
[debug] new argv=[/Users/roddie/Downloads/Rpi/balena/balena-cli/balena,/snapshot/volume/bin/balena,os:configure,balena-rpi4-image.img,--config,config.json,--device,6053dab8dc4721ed288c8dfc79e52967] length=8
? Network Connection ethernet
Configuring operating system image
IO: Input/output error

Error: Input/output error
    at Object.IO (/snapshot/volume/node_modules/fatfs/structs.js:251:17)
    at /snapshot/volume/node_modules/fatfs/chains.js:122:69
    at /snapshot/volume/node_modules/fatfs/vol.js:102:44
    at Object.<anonymous> (/snapshot/volume/node_modules/fatfs/chains.js:55:34)
    at Immediate._onImmediate (/snapshot/volume/node_modules/fatfs/helpers.js:168:12)
    at processImmediate (internal/timers.js:456:21)
    at process.topLevelDomainCallback (domain.js:137:15)
From previous event:
    at /snapshot/volume/node_modules/balena-device-init/build/network.js:125:14
    at _finish (/snapshot/volume/node_modules/fifolock/index.js:31:13)
    at /snapshot/volume/node_modules/fifolock/index.js:36:20
From previous event:
    at /snapshot/volume/node_modules/balena-device-init/build/network.js:117:25
    at runInFat (/snapshot/volume/node_modules/balena-image-fs/build/index.js:62:18)
    at async tryInteract (/snapshot/volume/node_modules/balena-image-fs/build/index.js:81:16)
    at async diskInteract (/snapshot/volume/node_modules/balena-image-fs/build/index.js:109:12)
    at async /snapshot/volume/node_modules/balena-image-fs/build/index.js:126:20
    at async Object.withOpenFile (/snapshot/volume/node_modules/file-disk/build/index.js:106:16)
    at async Object.interact (/snapshot/volume/node_modules/balena-image-fs/build/index.js:124:16)

For further help or support, visit:
https://www.balena.io/docs/reference/balena-cli/#support-faq-and-troubleshooting
```

### LINKED ISSUE #44: Needs revamp: apps need to be replaced by fleets
@klutchell The conversion from apps to fleets in v1.0.33 by Taro Murao is good, but by far was not complete.

Look for the remaining mentions of `apps`, `altApp` and `Application` in the document, examples:
```
$ BALENARC_BALENA_URL=balena-staging.com balena apps
ID    APP NAME          DEVICE TYPE  ONLINE DEVICES DEVICE COUNT
12345 appOne            intel-nuc    0              0
12346 appTwo            iot2000      0              0
...
Now try listing the apps from the staging environment again:

$ balena apps
ID    APP NAME          DEVICE TYPE  ONLINE DEVICES DEVICE COUNT
12345 appOne            intel-nuc    0              0
12346 appTwo            iot2000      0              0
...
$ cd ~/balenaStaging
$ balena apps
ID    APP NAME          DEVICE TYPE  ONLINE DEVICES DEVICE COUNT
12345 appOne            intel-nuc    0              0
12346 appTwo            iot2000      0              0

$ cd ~/balenaProduction
$ balena apps
ID      APP NAME         DEVICE TYPE      ONLINE DEVICES DEVICE COUNT
54321   artik530         artik533s        0              0
54322   bob              intel-nuc        0              0
54323   bbbtest          beaglebone-black 0              0
...
This allows multiple configs to be generated, for example for different Fleets, and then using copies of a bare OS image of a particular version to prepare images for each of those apps.
```

..on-interactive movement, simply pass the optional --application switch to the command with the relevant Fleet name:
```
$ cd ~/balenaProduction
$ balena device move 94095f8 --application altFleet
94095f8 was moved to altFleet
```

As you can see, there is a major undertaking left to update this masterclass.

### PR DESCRIPTION
**Pull request by rhampt**
Reworked this guide and verified the steps and output from each step and brought over to flowzone

Change-type: patch
Signed-off-by: Ryan Hampton (@rhampt) <rhampt@balena.io>

### CODE REVIEW

**Comment by rhampt** [comment]
cc @bernhardkaindl

**Comment by rhampt** [comment]
Oh I think this needs a flowzone conversion for CI. Is there a chance that you've done such a converstion before @vipulgupta2048

**Review by vipulgupta2048 — COMMENTED** [review]
I would make a seperate PR for the flowzone conversion

**Comment by rhampt** [comment]
@vipulgupta2048 Why is that? Seems process for process-sake. Technically, the flowzone stuff is inclusive of the "masterclass refresh" IMO (content + CI update). I didn't say masterclass "content" refresh. I could mean repo refresh :)

File: README.md [hunk 1]
```diff
@@ -1,103 +1,63 @@
...
-
-* [balena CLI Masterclass](https://github.com/balena-io/balena-cli-masterclass)
+To gain the most from this masterclass, we recommend that you first undertake the [balena CLI Masterclass](https://www.balena.io/docs/learn/more/masterclasses/cli-masterclass/)
 
 ## Introduction
 
-The balena Command Line Interface (balena CLI) utility consists of a number
-of commands that allow a user to develop, deploy and manage balena fleets,
-and devices.
+The balena Command Line Interface (balena CLI) utility consists of a number of commands that allow a user to develop, deploy, and manage balena fleets and devices.
```
Comment (vipulgupta2048) [inline README.md]:
```suggestion
The balena Command Line Interface (balena CLI) consists of a number of commands that allow a user to develop, deploy, and manage balena fleets and devices.
```

Comment (rhampt) [reply README.md]: Good call, agreed!

File: README.md [hunk 2]
```diff
@@ -1,103 +1,63 @@
...
 
-## Prerequisite Classes
+## Prerequisites
 
-This masterclass builds upon knowledge that has been taught in previous classes.
-To gain the most from this masterclass, we recommend that you first undertake
-the following masterclasses:
-
-* [balena CLI Masterclass](https://github.com/balena-io/balena-cli-masterclass)
+To gain the most from this masterclass, we recommend that you first undertake the [balena CLI Masterclass](https://www.balena.io/docs/learn/more/masterclasses/cli-masterclass/)
```
Comment (vipulgupta2048) [inline README.md]:
```suggestion
To gain the most from this masterclass, we recommend that you first go through the [balena CLI Masterclass](https://www.balena.io/docs/learn/more/masterclasses/cli-masterclass/)
```

**Comment by rhampt** [comment]
@vipulgupta2048 but on second thought, I guess I see the point. Maybe that PR should merge first, then I rebase this branch on master again. You like the history squashed after we iterate through it too,  correct?

File: README.md [hunk 3]
```diff
@@ -1,103 +1,63 @@
...
 
 ## Introduction
 
-The balena Command Line Interface (balena CLI) utility consists of a number
-of commands that allow a user to develop, deploy and manage balena fleets,
-and devices.
+The balena Command Line Interface (balena CLI) consists of a number of commands that allow a user to develop, deploy, and manage balena fleets and devices.
```
Comment (vipulgupta2048) [inline README.md]:
```suggestion
The balena Command Line Interface (balena CLI) consists of several commands that allow a user to develop, deploy, and manage balena fleets and devices.
```

File: README.md [hunk 4]
```diff
@@ -1,103 +1,63 @@
...
 
-The previous balena CLI masterclass covered some of the most common techniques
-and topics. This masterclass aims to build on top of that, introducing you to
-additional features that can be used to gain finer control over the
-provisioning, deployment and management of devices.
+The previous balena CLI masterclass covered some of the most common techniques and topics. This masterclass aims to build on top of that, introducing you to additional features that can be used to gain finer control over the provisioning, deployment and management, of devices.
```
Comment (vipulgupta2048) [inline README.md]:
```suggestion
The previous balena CLI masterclass covered some of the most common techniques and topics. This masterclass aims to build on top of that, introducing you to additional features that can be used to gain finer control over the provisioning, deployment, and management of devices.
```

File: README.md [hunk 5]
```diff
@@ -1,103 +1,63 @@
...
+- Update balenaOS programmatically
+- Preconfigure and preregister downloaded images
+- Preload downloaded images with your application code
 
-If you have any questions about this masterclass as you proceed through it,
-or would like clarifications on any of the topics raised here, please do
-raise an issue on the repository this file is contained in, or contact
-us on the [balena forums](https://forums.balena.io/) where we'll be
-delighted to answer your questions.
+If you have any questions about this masterclass as you proceed through it, or would like clarifications on any of the topics raised here, please do raise an issue on the repository this file is contained in, or contact us on the [balena forums](https://forums.balena.io/) where we'll be delighted to answer your questions.
```
Comment (vipulgupta2048) [inline README.md]: Link the repo?

File: README.md [hunk 6]
```diff
@@ -1,103 +1,63 @@
...
 
-The location of the repository that contains this masterclass and all associated
-code is
-[https://github.com/balena-io/balena-cli-advanced-masterclass](https://github.com/balena-io/balena-cli-advanced-masterclass).
+The location of the repository that contains this masterclass and all associated code is: [https://github.com/balena-io/balena-cli-advanced-masterclass](https://github.com/balena-io/balena-cli-advanced-masterclass).
```
Comment (vipulgupta2048) [inline README.md]: delete this line instead, and link the repo in the para above.

**Review by vipulgupta2048 — COMMENTED** [review]
For this repository/masterclass, it's already done but highly recommend not breaking the 80 character limit lines into paragraphs. It makes reviewing quite hard since I can't figure out what you have changed from the original content. If possible, please revert your changes to only the content you have actually changed and leave everything as it is.

**Comment by rhampt** [comment]
> For this repository/masterclass, it's already done but highly recommend not breaking the 80 character limit lines into paragraphs. It makes reviewing quite hard since I can't figure out what you have changed from the original content. If possible, please revert your changes to only the content you have actually changed and leave everything as it is.

@vipulgupta2048 Unfortunately, removing the arbitrary line breaks was one of the primary things I *wanted* to change with this PR. Whoever wrote the markdown used a careless method of entering a newline character between 70 and 100 characters into each row. Since all modern text editors and the web browsers themselves allow for word wrapping, and markdown isn't code, it's better to have a paragraph be a paragraph (of text) and let rendering handle overflows depending on the viewport of the reader. Besides making this one time review harder to parse, what's the reasoning behind having a character limit for markdown? I agree it's useful for code, if not arbitrary where the cutoff is.

File: README.md (hunk 5 above)
Comment (rhampt) [reply README.md]: Good call, will add

**Comment by vipulgupta2048** [comment]
Yes, please squash the commits @rhampt 
The line breaking could have been done to provide a clearer way of writing markdown but i agree with your reasoning. Flowzone in this PR is fine too.

**Comment by rhampt** [comment]
@vipulgupta2048 I squashed all my current commits into one and pushed it up!

File: README.md [hunk 7]
```diff
@@ -248,291 +178,232 @@ Logging in to balena-staging.com
...
 
 ```shell
 $ cd /tmp
 $ BALENARC_BALENA_URL=balena-staging.com \
   BALENARC_DATA_DIRECTORY=~/balenaStaging \
   balena whoami
+== ACCOUNT INFORMATION
+USERNAME: ryanh
+EMAIL:    ryanh@balena.io
+URL:      balena-cloud.com
```
Comment (vipulgupta2048) [inline README.md]: Is that the correct output? Above it the environment the command is targeting is the staging environment. 
Let me know, I could be wrong. 
```suggestion
URL:      balena-staging.com
```

**Review by vipulgupta2048 — COMMENTED** [review]
Haven't reviewed the content fully because of reasons mentioned above. Otherwise looks good, found just one thing here that looked odd. Merge it when ready 🚀

File: README.md (hunk 7 above)
Comment (rhampt) [reply README.md]: @vipulgupta2048 Good catch! This output shouldn't even be there I don't think.

**Comment by rhampt** [comment]
@balena-ci I self-certify!

**Pull request merged**